src/app/pages/storage/modules/pool-manager/utils/generate-vdevs/grouped-disks.class.ts

```diff
@@ -10,30 +10,34 @@ import { getDiskTypeSizeMap } from 'app/pages/storage/modules/pool-manager/utils
 export class GroupedDisks {
   private diskMap: DiskTypeSizeMap;
 
-  constructor(
-    private disks: UnusedDisk[],
-  ) {
+  constructor(private disks: UnusedDisk[]) {
     this.diskMap = getDiskTypeSizeMap(disks);
   }
 
   findSuitableDisks(category: PoolManagerTopologyCategory): UnusedDisk[] {
-    const matchingType = this.diskMap[category.diskType];
     let matchingTypeAndSize: UnusedDisk[] = [];
+    const ssdDisks = this.diskMap.SSD[category.diskSize] || [];
+    const hddDisks = this.diskMap.HDD[category.diskSize] || [];
+
     if (category.treatDiskSizeAsMinimum) {
-      for (const diskSize of Object.keys(matchingType)) {
+      for (const diskSize of Object.keys(this.diskMap.HDD)) {
+        if (category.diskSize > Number(diskSize)) continue;
+        matchingTypeAndSize = matchingTypeAndSize.concat(this.diskMap.HDD[diskSize]);
+      }
+      for (const diskSize of Object.keys(this.diskMap.SSD)) {
         if (category.diskSize > Number(diskSize)) continue;
-        matchingTypeAndSize = matchingTypeAndSize.concat(matchingType[diskSize]);
+        matchingTypeAndSize = matchingTypeAndSize.concat(this.diskMap.SSD[diskSize]);
       }
     } else {
-      matchingTypeAndSize = matchingType[category.diskSize];
+      matchingTypeAndSize = Array.from(
+        new Map([...hddDisks, ...ssdDisks].map((value) => [value.devname, value])).values(),
```

Comment:
I don't think it's a good idea to return one property when method declares to return the whole interface.

Reply:
in the end it's an array of data with same interface, we call `.values()` to receive values from Map by key.